crates/bevy_pbr/src/render/mesh_view_bindings.wgsl

```diff
@@ -42,12 +42,31 @@
 @group(0) @binding(16) var dt_lut_sampler: sampler;
 
 #ifdef MULTISAMPLED
+
+#ifdef DEPTH_PREPASS
 @group(0) @binding(17) var depth_prepass_texture: texture_depth_multisampled_2d;
+#endif // DEPTH_PREPASS
+#ifdef NORMAL_PREPASS
 @group(0) @binding(18) var normal_prepass_texture: texture_multisampled_2d<f32>;
+#endif // NORMAL_PREPASS
+#ifdef MOTION_VECTOR_PREPASS
 @group(0) @binding(19) var motion_vector_prepass_texture: texture_multisampled_2d<f32>;
-#else
+#endif // MOTION_VECTOR_PREPASS
+
+#else // MULTISAMPLED
+
+#ifdef DEPTH_PREPASS
 @group(0) @binding(17) var depth_prepass_texture: texture_depth_2d;
```

Comment:
Does it work to skip binding numbers like this? I assume so.

Reply:
What do you mean by skipping? Everything looks fine to me here?

Reply:
Hmm I _assumed_ it would work (After all, why would the spec include indexes if they must always be sequential) and it _is_ working on both Metal, WebGL and WebGPU backends, but I wasn't sure, so I double checked it in the spec. They're quite roundabout about it but they do mention the following:

<img width="745" alt="image" src="https://github.com/bevyengine/bevy/assets/418473/6b0473b1-336e-47e7-940c-e42cee2e34b8">
<img width="838" alt="image" src="https://github.com/bevyengine/bevy/assets/418473/4fa37101-60ff-4f76-94b0-286eb43c44f2">
<img width="855" alt="image" src="https://github.com/bevyengine/bevy/assets/418473/eee2321b-94f9-40b1-b341-c4c6799d012f">

So layout entry indexes:

- Are "unique identifiers",
- Can go up to 999 (despite it not being possible to have 999 bindings in one group)
- Have their limit set based on the implementation not needing to use sparse data structures
- Have their compatibility checked via set-like, non contiguous semantics

So yeah it should be 100% safe to omit some of them

Reply:
Oh, right, some of them would skip if they are behind a shader defs.

Yeah, I've done that before and it worked without issues.